### ISSUE TITLE
MockFileSystem.Directory.EnumerateFiles does not return files with no extension when Search Pattern is "*."

### ISSUE THREAD

**Issue by ptgullas** [System-IO-Abstractions/System.IO.Abstractions#439]
When using a MockFileSystem, Directory.EnumerateFiles doesn't return files that don't have an extension, and the search pattern is "*." . It works with a real file system.

The below test fails:
``` c#
        [TestMethod]
        public void MockFileSystem_EnumerateFiles_ShouldReturnFilesWithNoExtensions() {
            var fs = new MockFileSystem();
            var mf = new MockFileData("my name is Luka");
            string filePath = @"c:\downloadfolder\ThisIsAFileWithNoExtension";
            fs.AddFile(filePath, mf);
            IEnumerable<string> files = fs.Directory.EnumerateFiles(@"c:\downloadfolder", "*.", SearchOption.TopDirectoryOnly);
            Assert.AreEqual(1, files.Count());
        }
```

**Comment by fgreinacher** [comment]
Thanks for reporting!

**Comment by fgreinacher** [comment]
Fixed with #516
